modules/openapi-generator/src/main/resources/cpp-qt5-client/HttpRequest.cpp.mustache

```diff
@@ -368,18 +376,61 @@ void {{prefix}}HttpRequestWorker::process_form_response() {
                 }
             }
             {{prefix}}HttpFileElement felement;
-            felement.saveToFile(QString(), workingDirectory + QDir::separator() + filename, filename, contentType, response.data());
+            felement.saveToFile(QString(), workingDirectory + QDir::separator() + filename, filename, contentType, reply->readAll());
             files.insert(filename, felement);
         }
 
     } else if (getResponseHeaders().contains(QString("Content-Type"))) {
         auto contentType = getResponseHeaders().value(QString("Content-Type").toUtf8()).split(QString(";"), QString::SkipEmptyParts);
         if ((contentType.count() > 0) && (contentType.first() == QString("multipart/form-data"))) {
+            // TODO : Handle Multipart responses
         } else {
+            if(headers.contains("Content-Encoding")){
+                auto encoding = headers.value("Content-Encoding").split(',');
+                if(encoding.contains("gzip", Qt::CaseInsensitive) || encoding.contains("deflate", Qt::CaseInsensitive)){
+                    response = decompress(reply->readAll());
+                }
+            }
+            else {
+                response = reply->readAll();
+            }
         }
     }
 }
 
+QByteArray {{prefix}}HttpRequestWorker::decompress(const QByteArray& data){
+    QByteArray result;
+    bool sts = false;{{#contentCompression}}
+    do{
+        z_stream strm;
```

Comment:
wouldn't it be better to zero-initialize strm?

```
z_stream strm{};
```

with this you could get rid of
```
        strm.zalloc = Z_NULL;
        strm.zfree = Z_NULL;
        strm.opaque = Z_NULL;
```